www/ec2/ec2.inc.php

```diff
@@ -593,11 +597,48 @@ function EC2Log($msg) {
 
 /**
 * Log an error to both the EC2 log and the error log
-* 
+*
 * @param mixed $msg
 */
 function EC2LogError($msg) {
   EC2Log('Error: ' . $msg);
   logError('EC2:' . $msg);
 }
-?>
\ No newline at end of file
+
+/**
+ * A tag delimited string looks as follows:
+ *   'k1=>v1|k2=>v2|k3=>v3'
+ * And this function will return the following:
+ *  Array (
+ *     [0] => Array (
+ *       [Key] => k1
+ *       [Value] => v1
+ *     )
+ *
+ *     [1] => Array (
+ *       [Key] => k2
+ *       [Value] => v2
+ *     )
+ *
+ *     [2] => Array (
+ *       [Key] => k3
+ *       [Value] => v3
+ *     )
+ *   )
+ *
+ * We use hash rockets and pipes to build our string because
+ * they are much less likely to be used in a tag than other characters.
+ * @param string $tagstring A tag delimited string.
+ * @return array An array of array tag key-value pairs
+ */
+function EC2_CreateTagArray($tagstring) {
+  $kvpairs = explode('|', $tagstring);
+  $final_array = array();
+
+  foreach ($kvpairs as $kvpair) {
+    $pair = explode('=>', $kvpair);
+    array_push($final_array, array('Key' => $pair[0], 'Value' => $pair[1]));
```

Comment:
nit: could you make this:
$final_array[] = array('Key' => $pair[0], 'Value' => $pair[1]);

Everything else looks great.